src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -661,16 +661,31 @@ public static boolean compareRow(List<Object> expectedRow, ResultSet actualRow)
                     break;
                 case Types.OTHER:
                     Object expected_obj = expectedRow.get(i);
-                    Object actual_obj = actualRow.getObject(i + 1);
-                    if (!expected_obj.equals(actual_obj)) {
-                        System.err.println(
-                                "Expected Bson Other value "
-                                        + expected_obj
-                                        + " but is "
-                                        + actual_obj
-                                        + " for column "
-                                        + (i + 1));
-                        return false;
+                    if (expected_obj instanceof String) {
```

Comment:
Added this branch to use getString() instead of getObject() if the expected input is a string.  Which will be from an `expected_result` entry as opposed to `expected_result_extended_json` which wouldn't have a String